### DISCUSSION TITLE
Remote deletion of a template is not reflected in template listing in CP

### DISCUSSION THREAD

**Discussion by jcogs-design** [ExpressionEngine/ExpressionEngine#2938] [Feature Requests]
If you delete a template in the (e.g.) default_site template directory, EE does not update its internal records to match, and will restore the deleted template when a refresh of a CP template view is made.

This is a change to previous behaviour - where deleting a template file externally would cause the file to be deleted within the template records in the EE database too.

Edits to the content of a template, and additions of new template files to the directories are correctly recognised and reflected in the CP view of templates.

I am unsure when this issue arose - it is present in both EE 5.4.1 and EE 6.0.2 but I only spotted when some old templates that used an add-on that had been removed showed up after the EE 6.0.2 update: the templates had been deleted some months before from the remote copy of the site used to maintain templates; am guessing they were not deleted (and then restored) by EE when the template directory was updated at the time.

Being unable delete template files remotely introduces a tedious workflow issue - you have to manually delete the template groups affected from the EE CP before uploading the corrected template set (and reset the default group afterwards) to achieve the correct update.

**Comment by intoeetive** [comment]
I believe that in EE templates saved to database are to be considered the source of truth, as all operations as based on their ID in `exp_templates`.

One can use files to update templates, or even create new ones, but when it comes to deletion - you can only delete from control panel.

I can't speak for sure regarding some older EE version, but this behaviour has been there for a while already.

@robinsowell might have more to say on this

**Comment by jcogs-design** [comment]
Well maybe. Perhaps it is not so visible usually as often the number of template deletes is small. The "EE database is the master" model certainly applied in EE2 (when 'synchronising' templates with external files was an optional activity), but I was under the impression that with either EE4 or 5 the assumed master became the files.  But maybe I'm mistaken and this dysfunctional behaviour has been the model all the way through - I need to go back and check my sites' template directories for garbage files if so :(

AFAIK most EE template developers work using files rather than working live in CP. The implied requirement of the "no delete" state is for developers to have to continually manually reconcile the templates in the EE database with those in the templates directory - a requirement that works against good practices (as it is likely to result in garbage templates accumulating in the EE database) and undermines the utility of supporting file-based development.

If this is not something that should be fixed, perhaps this should become a feature request.

**Comment by robinsowell** [comment]
I do think it makes sense as a feature request.  We wouldn't change how it works now, but perhaps a setting would work.  Make it so if a file does not exist, it deletes the db record.  But I definitely think changing the current behavior would be problematic, and we'd want to go with a setting.
